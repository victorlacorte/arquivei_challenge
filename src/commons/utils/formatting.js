function toBRL(number) {
  return number.toLocaleString('pt-BR', { style: 'currency', currency: 'BRL' });
}

export {
  toBRL,
};
