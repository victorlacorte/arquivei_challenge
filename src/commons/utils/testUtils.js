/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */
import React from 'react';
import { ThemeProvider } from 'styled-components';
// eslint-disable-next-line import/no-extraneous-dependencies
import { mount, shallow } from 'enzyme';

import defaultTheme from 'commons/styles/theme';

function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>
        {children}
      </ThemeProvider>
    ),
  });
}

function shallowWithTheme(child) {
  return shallow(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>
        {children}
      </ThemeProvider>
    ),
  });
}

export {
  mountWithTheme,
  shallowWithTheme,
};
