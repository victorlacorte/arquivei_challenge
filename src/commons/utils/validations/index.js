export { default as cnpjValidations } from './cnpj';
export { default as consultKeysValidations } from './consultKeys';
export { default as creditCardValidations } from './creditCard';
export { default as formValidations } from './formValidations';
