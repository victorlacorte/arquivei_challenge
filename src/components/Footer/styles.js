import { memo } from 'react';
import styled from 'styled-components';

/**
 * Total footer height:
 * Desktop: 60px
 * Mobile: 40px
 *
 * This information could be in the theme since our Main component makes use of it
 */
const StyledFooter = styled.footer`
  background-color: ${({ theme }) => theme.palette.primary.main};
  box-shadow: 0 -3px 5px ${({ theme }) => theme.palette.grey400.main};
  height: 60px;
  z-index: ${({ theme }) => theme.zIndex.appBar};

  ${({ theme }) => theme.breakpoints.down('sm')`
    height: 40px;
  `}
`;

export default memo(StyledFooter);
