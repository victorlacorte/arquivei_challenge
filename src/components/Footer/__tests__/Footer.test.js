import React from 'react';

import Footer from 'components/Footer';
import { shallowWithTheme } from 'commons/utils/testUtils';

describe('<Footer />', () => {
  it('creates a Footer component', () => {
    const wrapper = shallowWithTheme(<Footer />);

    expect(wrapper).toMatchSnapshot();
  });
});
