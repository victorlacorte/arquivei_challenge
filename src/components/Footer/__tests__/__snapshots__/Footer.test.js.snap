// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> creates a Footer component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "styles__StyledFooter-a3uq45-0",
        "isStatic": false,
        "rules": Array [
          "background-color:",
          [Function],
          ";box-shadow:0 -3px 5px ",
          [Function],
          ";height:60px;z-index:",
          [Function],
          ";",
          [Function],
        ],
      },
      "displayName": "styles__StyledFooter",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "styles__StyledFooter-a3uq45-0",
      "target": "footer",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
